libs/dal/src/repositories/notification/notification.repository.ts

```diff
@@ -114,6 +114,7 @@ export class NotificationRepository extends BaseRepository<EnforceEnvironmentQue
           _environmentId: new Types.ObjectId(environmentId),
         },
       },
+      { $unwind: '$channels' },
```

Comment:
the channels is an array, we should unwind it to be able to pass all items to the set